### DISCUSSION TITLE
Calculating Jacobian with respect to certain outputs only

### DISCUSSION THREAD

**Discussion by Justin-Tan** [jax-ml/jax#8820] [Q&A] [answered]
Hi, is it possible to calculate the Jacobian of a function with respect to certain outputs only? 

e.g. for 
```python
def my_func(x,y,z):
    return x**2,y**2,z**2
```
I'd like to find the Jacobian of the first output only, with respect to the function's input arguments.

**Comment by jakevdp** [comment, marked as answer]
I think the recommended way to do this would be something like
```python
my_func_jac = jax.jacobian(lambda x, y, z: my_func(x, y, z)[0])
```
If you `jit`-compile the function, the XLA compiler will do dead code elimination to ensure that unused results are not computed.

**Reply by Justin-Tan** [reply]
Thanks! Would using a `functools.partial` call over the `lambda` achieve the same thing?

**Reply by jakevdp** [reply]
I'm not sure what you mean
